server/src/main/java/org/opensearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -188,6 +188,20 @@ public void recoverToTarget(ActionListener<RecoveryResponse> listener) {
         }
     }
 
+    protected void preventRefreshOnReplicas() {
+        // Disable this shard from performing segment replication on replicas
+        if (shard.indexSettings().isSegRepEnabled() && request.isPrimaryRelocation() == true) {
+            shard.setBlockInternalCheckPointRefresh(true);
+        }
+    }
+
+    protected void resumeSegmentReplicationRefresh() {
+        // Enable this shard from performing segment replication on replicas
+        if (shard.indexSettings().isSegRepEnabled() && request.isPrimaryRelocation() == true) {
+            shard.setBlockInternalCheckPointRefresh(false);
```

Comment:
Can we assert if the previous state was blocked before unblocking

Reply:
Updated.